buildtest/modules/util.py

```diff
@@ -149,8 +177,14 @@ def describe(self, collection="default"):
         module_describe_cmd = f"module describe {collection}"
         cmd = BuildTestCommand()
         cmd.execute(module_describe_cmd)
+        ret = cmd.returnCode()
 
         # For some odd reason, the output of module save gets passed to STDERR stream instead of STDOUT.
         out = cmd.get_error()
 
+        # print executed command for debugging
+        if self.debug:
```

Comment:
This is a lot of redundant code - can this be a shared function for the class? I still think there could be a very basic base class for ModuleCollection and Module, or at least a shared function added to both to essentially run a command and print debug output (or not).

Reply:
I think it makes sense to have a shared method for debug purpose.

It will require some code refactor and changes to example program. 

Id be open to changing the ModuleCollection class. I think this PR is already big enough.

I'd eventually like to have this part of regression test. In subsequent PRs.

Also Travis builds a mini eb stack for testing some of these features. That's a whole separate discussion

Reply:
For this kind of debugging I usually put the logger in charge of this - you import the logger and then can just do logger.debug / logger.info and it will show (or not) depending on the user selected level.